src/go/rdctl/pkg/factoryreset/factory_reset_windows.go

```diff
@@ -22,18 +22,129 @@ import (
 	"os"
 	"os/exec"
 	"path"
+	"path/filepath"
+	"sort"
 	"strings"
 	"syscall"
+	"unsafe"
 
 	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/directories"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows"
 	"golang.org/x/text/encoding/unicode"
 )
 
-func KillRancherDesktop() {
-	cmd := exec.Command("taskkill", "/IM", "Rancher Desktop.exe", "/T", "/F")
-	cmd.SysProcAttr = &syscall.SysProcAttr{CreationFlags: CREATE_NO_WINDOW}
-	cmd.Run()
+var (
+	pKernel32      = windows.NewLazySystemDLL("kernel32.dll")
+	pEnumProcesses = pKernel32.NewProc("K32EnumProcesses")
+)
+
+func KillRancherDesktop() error {
```

Comment:
This function could use a high-level comment explaining its main purpose and what it does, with more comments on details in line (which I'll point out)

Reply:
Added. In the process of writing the comment, I realized that I really just want to exclude the current process, not all `rdctl` processes; so I did that instead.